tests/lib/rules/no-unused-prop-types.js

```diff
@@ -2012,6 +2012,79 @@ ruleTester.run('no-unused-prop-types', rule, {
         '}'
       ].join('\n'),
       parser: 'babel-eslint'
+    }, {
+      // issue #1002
+      code: [
+        '  type User = {',
+        '    country: string;',
+        '  };',
+
+        '  export default ({ user }: { user: User }) => (',
+        '    <div>',
+        '      { user.country }',
+        '    </div>',
+        '  );'
+      ].join('\n'),
+      parser: 'babel-eslint'
+    }, {
+      // issue #1008
+      code: [
+        'type ObjectType = {',
+        '  usedDirectly: string;',
+        '  usedFromArray: string;',
```

Comment:
shouldn't this example only pass when skipShapeProps is false, but fail when it's true?

If so, could we add an explicit config option?